### DISCUSSION TITLE
Cannot verify email after registering

### DISCUSSION THREAD

**Discussion by joshembling** [Jacobtims/filament-breezy#169] [Q&A]
I have a user set up like so: 

```
class User extends Authenticatable implements FilamentUser, MustVerifyEmail
{
    use HasApiTokens, HasFactory, Notifiable;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'name',
        'email',
        'password',
    ];

    /**
     * The attributes that should be hidden for serialization.
     *
     * @var array<int, string>
     */
    protected $hidden = [
        'password',
        'remember_token',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'email_verified_at' => 'datetime',
        'password' => 'hashed',
    ];

    public function canAccessFilament(): bool
    {
        return $this->hasVerifiedEmail();
    }
}
```

Few issues:
1. On registering at `admin/register` I get redirected to my home route, currently `/`, not the 'verify' view. 
2. Email to verify is successfully sent but I can't access the verification link, so as long as I have the FilamentUser interface on my User model with the `canAccessFilament()` method.

Obviously I only want verified users to access my admin panel, but I need to access the verification route to make that happen first before I `email_verified_at` gets updated.. 

Seems that is happening in EmailVerificationController, but I can't get past the construct. 

```
class EmailVerificationController extends Controller
{
    public function __construct()
    {
        $this->middleware([Authenticate::class]);
    }

```

TLDR - How can I add constraints to `canAccessFilament()` but still allow users to verify emails before accessing admin panel?
